coordinator/internal/logic/provertask/batch_prover_task.go

```diff
@@ -154,13 +138,25 @@ func (bp *BatchProverTask) Assign(ctx *gin.Context, getTaskParameter *coordinato
 	}
 
 	log.Info("start batch proof generation session", "id", batchTask.Hash, "public key", taskCtx.PublicKey, "prover name", taskCtx.ProverName)
+	var (
+		proverVersion = taskCtx.ProverVersion
+		hardForkName  = taskCtx.HardForkName
+	)
+	var err error
+	if getHardForkName != nil {
```

Comment:
can add a check about `taskCtx.HardForkName` to separate two cases.

Reply:
I wonder whether or not it's necessary to check the taskCtx.HardForkName here. Since there are only two code paths calling this function, and there shall be no ambiguity.

Reply:
make sense.